dlib/image_transforms/interpolation.h

```diff
@@ -252,72 +253,18 @@ namespace dlib
             const double lr_frac = p.x() - left;
             const double tb_frac = p.y() - top;
 
-            double tl = 0, tr = 0, bl = 0, br = 0;
-
-            assign_pixel(tl, img[top][left]);
-            assign_pixel(tr, img[top][right]);
-            assign_pixel(bl, img[bottom][left]);
-            assign_pixel(br, img[bottom][right]);
-            
-            double temp = (1-tb_frac)*((1-lr_frac)*tl + lr_frac*tr) + 
-                              tb_frac*((1-lr_frac)*bl + lr_frac*br);
-                            
-            assign_pixel(result, temp);
+            auto const tl = pixel_to_vector<typename traits::basic_pixel_type>(img[top][left]);
+            auto const tr = pixel_to_vector<typename traits::basic_pixel_type>(img[top][right]);
+            auto const bl = pixel_to_vector<typename traits::basic_pixel_type>(img[bottom][left]);
+            auto const br = pixel_to_vector<typename traits::basic_pixel_type>(img[bottom][right]);
```

Comment:
`const auto tl = pixel_to_vector<double>(img[top][left])` would be simpler and better here, since the interpolation needs to happen in floating point anyway, and this makes it clear that it does.  It's also easier to read.

Reply:
Just checked that performance of this should be good as expected (putting temporary into array should not impact performance).
https://godbolt.org/z/wXt9CM
I intend to do quadratic too with shorter code.